pkg/controller/handlers/mcpcatalog/mcpcatalog.go

```diff
@@ -362,8 +362,7 @@ func (h *Handler) DeleteUnauthorizedMCPServerInstances(req router.Request, _ rou
 
 	// Iterate through each MCPServerInstance and make sure it is still allowed to exist.
 	for _, instance := range mcpServerInstances.Items {
-		if !instance.DeletionTimestamp.IsZero() {
-			// This server is already being deleted, so we don't have to delete it again.
+		if instance.Spec.MCPCatalogName != system.DefaultCatalog || !instance.DeletionTimestamp.IsZero() {
```

Comment:
This is future-proofing by skipping the loop iteration if the instance is outside the default catalog. This cannot happen currently, since everything is in the default catalog.

Reply:
Is there a reason why we would want to skip this logic if the instance is outside of the default catalog? I don't see any logic in the loop that would break in that case.

Reply:
Well, only because the ACRs are not catalog scoped and we only apply them to the default catalog. Obviously it does not matter right now, but that's why I called this future-proofing, but I think I am convinced that there's not a point in doing this, so I'll just remove it.

Reply:
The access control rules not being catalog scoped was the part I was missing. Thanks for that clarification.